### PR TITLE
Convert globalstatedb package to a standard store

### PR DESCRIPTION
**Pull request by camdencheek**
This converts the `globalstatedb` into a more standard store. Each of the 
functions in the `globalstatedb` package are converted to methods on the
new `GlobalStateStore` type. This allows us to generate mocks for the store,
getting rid of the global package-level mocks, and also allows us to get rid of
a few more instances of `dbconn.Global` by passing in a database from a
parent context.

Stacked on #28120 

3 down, 34 to go.

### CODE REVIEW

File: internal/database/dbtesting/dbtesting.go [hunk 1]
```diff
@@ -51,9 +51,9 @@ var (
 	connectErr  error
 )
 
-// SetupGlobalTestDB creates a temporary test DB handle, sets
+// setupGlobalTestDB creates a temporary test DB handle, sets
 // `dbconn.Global` to it and setups other test configuration.
-func SetupGlobalTestDB(t testing.TB) {
+func setupGlobalTestDB(t testing.TB) {
```
Comment (camdencheek) [inline internal/database/dbtesting/dbtesting.go:56]: This is no longer used outside the `dbtesting` package!

Comment (mrnugget) [reply internal/database/dbtesting/dbtesting.go:56]: Nice!! Finally!

File: yarn.lock [hunk 2]
```diff
@@ -423,7 +423,7 @@
   resolved "https://registry.npmjs.org/@babel/parser/-/parser-7.11.5.tgz#c7ff6303df71080ec7a4f5b8c003c58f1cf51037"
```
Comment (mrnugget) [inline yarn.lock:423]: Does this file need to be in this PR? (Just curious)

**Review by mrnugget — APPROVED** [review]

**Review by unknwon — APPROVED** [review]

File: yarn.lock (hunk 2 above)
Comment (camdencheek) [reply yarn.lock:423]: Hmm. No, it probably shouldn't be. My setup keeps modifying that file, and I'm not sure why

**Pull request merged**